### PR TITLE
Saving image GID in the discount 

### LINKED ISSUE #28: Add error handling in discount form submission
The errors in discount form submission silently fail. It should return meaningful messages to the user.

### PR DESCRIPTION
**Pull request by MiroBenicio**
# Issues Addressed
Closes https://github.com/tiki/apps-shopify-extensions/issues/43

## Additional Information
Now its saving the GID of the MediaImage into the discount data

### CODE REVIEW

File: src/_worker.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ router
   .get(`${API.Consts.API_LATEST}/shop/uninstall`, Shop.uninstall)
   .post(`${API.Consts.API_LATEST}/discount`, Discount.create)
   .get(`${API.Consts.API_LATEST}/discount/:id`, Discount.get)
+  .post(`${API.Consts.API_LATEST}/upload/stage`, Discount.stagedUpload)
```
Comment (ricardolg-dev) [inline src/_worker.ts]: since it is all done in the same call just use "upload"

File: src/components/BannerImageDescription.tsx [hunk 2]
```diff
@@ -37,7 +38,13 @@ export const BannerImageDescription = () => {
 
   return (
     <>
-        <DropZone allowMultiple={false} onDrop={handleDropZoneDrop}
+        <DropZone allowMultiple={false} onDrop={
+          handleDropZoneDrop
+        }
+        onDropAccepted={(file)=>{
+          console.log('teste componente', file)
```
Comment (ricardolg-dev) [inline src/components/BannerImageDescription.tsx]: remove logs

File: src/components/BannerImageDescription.tsx [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
     [],
   );
 
   const validImageTypes = ['image/gif', 'image/jpeg', 'image/png'];
-
   const fileUpload = !bannerFile && <DropZone.FileUpload />;
   const uploadedFile = bannerFile && (
     <LegacyStack>
-      <Thumbnail
+      {/* <Thumbnail
```
Comment (ricardolg-dev) [inline src/components/BannerImageDescription.tsx]: remove comment and add story to put thumbnail

File: src/components/providers/AppBridgeProvider.tsx [hunk 4]
```diff
@@ -25,7 +25,7 @@ export function AppBridgeProvider(props: { children: React.ReactNode }) {
     [history, location],
   );
 
-  const apiKey = '33d82ccecd1a316a4cbdb7d090735fa8'; // process.env.SHOPIFY_API_KEY
+  const apiKey = '1949a2edb1cab0db98cdf67e81742982'; // process.env.SHOPIFY_API_KEY
```
Comment (ricardolg-dev) [inline src/components/providers/AppBridgeProvider.tsx]: rollback. Create story to use env vars

File: src/pages/discount/order/Create.tsx [hunk 5]
```diff
@@ -184,15 +118,17 @@ if (response.ok) {
         productDiscounts: false,
         shippingDiscounts: combinesWith.shippingDiscounts,
       },
+      discountImg: imageId,
+      discountDescription: offerDescription
     };
     await authenticatedFetch(
-      'https://intg-shpfy.pages.dev/api/latest/discount',
+      'https://tiki-web.pages.dev/api/latest/discount',
```
Comment (ricardolg-dev) [inline src/pages/discount/order/Create.tsx]: rollback

Comment (ricardolg-dev) [reply src/pages/discount/order/Create.tsx]: check if it is possible to use request URL to build it.

File: src/pages/discount/product/Create.tsx [hunk 6]
```diff
@@ -207,15 +141,15 @@ await collectionUpdate({ variables: {
...
-    await authenticatedFetch(
-      'https://intg-shpfy.pages.dev/api/latest/discount',
-      {
-        method: 'POST',
-        headers: { 'Content-Type': 'application/json' },
-        body: JSON.stringify(body),
-      },
-    );
+    console.log('body:', body)
+    await authenticatedFetch('https://tiki-web.pages.dev/api/latest/discount', {
```
Comment (ricardolg-dev) [inline src/pages/discount/product/Create.tsx]: rollback

**Review by ricardobrg — CHANGES_REQUESTED** [review]

**Review by ricardobrg — APPROVED** [review]

**Pull request merged**